### PR TITLE
feat(component/List): add virtualized

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

The list is slow when lot's of items in it.

**What is the chosen solution to this problem?**

The list accept a new props 'virtualized' which trigger the use of VirtualizedList
with a compatibility mode.

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**state**

This PR is ready to be used in some project like TFD. Some missing binding are left to other teams:

- support for selection
- any other ?

### CODE REVIEW

File: packages/components/src/List/ListToVirtualizedList/ListToVirtualizedList.component.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+function ListToVirtualizedList(props) {
+	const { id, items, columns, titleProps } = props;
+	if (!titleProps.actionsKey) {
+		titleProps.actionsKey = 'actions';
+	}
+	// Backward compatibility: find array in object attr:
+	const supposedActions = {};
+	if (props.items.length > 0) {
+		const item = props.items[0];
+		Object.keys(item).forEach((key) => {
```
Comment (talend-glorieux) [inline packages/components/src/List/ListToVirtualizedList/ListToVirtualizedList.component.js]: I'd rather see a reduce used here.

File: packages/containers/src/List/List.container.js [hunk 2]
```diff
@@ -102,6 +102,7 @@ class List extends React.Component {
 				items,
 				columns: get(this.props, 'list.columns', []),
 			},
+			virtualized: this.props.virtualized || false,
```
Comment (jsomsanith) [inline packages/containers/src/List/List.container.js]: Remove `|| false`, not needed (+ update snapshots)

File: packages/components/src/List/ListToVirtualizedList/ListToVirtualizedList.component.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+					<VirtualizedList.Content key={index} {...cProps} />
+				);
+			})}
+		</VirtualizedList>
+	);
+}
+
+ListToVirtualizedList.propTypes = {
+	id: PropTypes.string,
+	displayMode: PropTypes.oneOf(['LARGE', 'TABLE']),
```
Comment (jsomsanith) [inline packages/components/src/List/ListToVirtualizedList/ListToVirtualizedList.component.js]: Warning on the storybook : 
```
warning.js:36 Warning: Failed prop type: Invalid prop `displayMode` of value `large` supplied to `ListToVirtualizedList`, expected one of ["LARGE","TABLE"].
```

**Comment by jlame** [comment]
Why not add some rows (like a lot)  to the story book as the list is now virtualize ?

File: packages/components/src/List/ListToVirtualizedList/ListToVirtualizedList.component.js (hunk 1 above)
Comment (jsomsanith) [inline packages/components/src/List/ListToVirtualizedList/ListToVirtualizedList.component.js]: Reduce in this case is not really improving the readability. But we still can improve that : 
```
Object.keys(item)
	.filter(key => Array.isArray(item[key]))
	.forEach((key) => { supposedActions[key] = true; });
``

**Pull request merged**